### ISSUE TITLE
the trait bound `*const str: std::marker::Sync` is not satisfied in `string_interner::InternalStrRef`

### ISSUE THREAD

**Issue by messense** [Robbepop/string-interner#1]
I am trying to use this crate with [Rocket](https://rocket.rs), but got the error:

```
error[E0277]: the trait bound `*const str: std::marker::Send` is not satisfied in `string_interner::InternalStrRef`
  --> src/main.rs:33:1
   |
33 |   fn show_graph_info(graph: State<PathGraphState>) -> JSON<Value> {
   |  _^ starting here...
34 | |     JSON(json!({
35 | |         "count": {
36 | |             "nodes": graph.0.graph.node_count(),
37 | |             "edges": graph.0.graph.edge_count()
38 | |         }
39 | |     }))
40 | | }
   | |_^ ...ending here: within `string_interner::InternalStrRef`, the trait `std::marker::Send` is not implemented for `*const str`
   |
   = note: `*const str` cannot be sent between threads safely
   = note: required because it appears within the type `string_interner::InternalStrRef`
   = note: required because of the requirements on the impl of `std::marker::Send` for `std::collections::hash::table::RawTable<string_interner::InternalStrRef, usize>`
   = note: required because it appears within the type `std::collections::HashMap<string_interner::InternalStrRef, usize>`
   = note: required because it appears within the type `string_interner::StringInterner<usize>`
   = note: required because it appears within the type `PathGraph`
   = note: required because it appears within the type `PathGraphState`
   = note: required by `rocket::State`
```

Is it possible to use it with Rocket?

**Comment by Robbepop** [comment]
Hello messense,
thank you very much for filing this issue!
I will take a look into it and try to fix this if possible or try to create another way to make this possible for your usage.

**Comment by Robbepop** [comment]
@messense Had a discussion about the safety of a `Send` and `Sync` implementation for `StringInterner` and the conclusion was that it should be safe. Fixed and published v0.3.3 on crates.io that now features `StringInterner` being `Send` + `Sync` if generic `Sym` is `Send` + `Sync`. ;)
